integration/run/run_test.go

```diff
@@ -1603,8 +1603,7 @@ func TestEnforcedQuota(t *testing.T) {
 	// Wait for the app to set the AppInstanceQuotaAllocated condition to be transitioning and for the namespace
 	// to be ready.
 	helper.WaitForObject(t, helper.Watcher(t, c), &apiv1.AppList{}, app, func(obj *apiv1.App) bool {
-		return obj.Status.Condition(v1.AppInstanceConditionNamespace).Success &&
-			obj.Status.Condition(v1.AppInstanceConditionQuotaAllocated).Transitioning
```

Comment:
My theory is that both conditions are only true transiently, and not eventually; i.e. we can start watching _too late_ and missed when both are true.